xbmc/addons/kodi-addon-dev-kit/CMakeLists.txt

```diff
@@ -2,6 +2,17 @@ project(kodi-addon-dev-kit)
 
 cmake_minimum_required(VERSION 2.8.7)
 
+# add a target to generate API documentation with Doxygen
+find_package(Doxygen)
+
+if(DOXYGEN_FOUND)
+  add_custom_target(doc
+    ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/doxygen/Doxyfile
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Generating API documentation with Doxygen" VERBATIM
+  )
+endif(DOXYGEN_FOUND)
+
```

Comment:
Yes exactly. It is then exposed as typical python docstrings (of limited value since you cannot run an interactive shell against kodi but still..)

-------- Opprinnelig melding --------
Fra: Alwin Esch <notifications@github.com>
Dato: 07.04.2016 06:08 (GMT+01:00)
Til: xbmc/xbmc <xbmc@noreply.github.com>
Ko: Arne Morten Kvarving <Arne.Morten.Kvarving@sintef.no>
Emne: Re: [xbmc/xbmc] [addon] binary add-on library rework for API level system (#9368)


In xbmc/addons/kodi-addon-dev-kit/CMakeLists.txt<https://github.com/xbmc/xbmc/pull/9368#discussion_r58816987>:

> @@ -2,6 +2,17 @@ project(kodi-addon-dev-kit)
>
>  cmake_minimum_required(VERSION 2.8.7)
>
> +# add a target to generate API documentation with Doxygen
> +find_package(Doxygen)
> +
> +if(DOXYGEN_FOUND)
> +  add_custom_target(doc
> +    ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/doxygen/Doxyfile
> +    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
> +    COMMENT "Generating API documentation with Doxygen" VERBATIM
> +  )
> +endif(DOXYGEN_FOUND)
> +


Thanks, you mean to have it here, on e.g. AddonModuleXbmcgui.cpp?

  PyDoc_STRVAR(XBMCAddon_xbmcgui_ListItem_setArt__doc__,
               "setArt(values) -- Sets the listitem's art \n"
"\n"
"values : dictionary - pairs of { label: value }.\n"
"\n"
"    - Some default art values (any string possible):\n"
"        - thumb : string - image filename\n"
"        - poster : string - image filename\n"
"        - banner : string - image filename\n"
"        - fanart : string - image filename\n"
"        - clearart : string - image filename\n"
"        - clearlogo : string - image filename\n"
"        - landscape : string - image filename\n"
"        - icon : string - image filename\n"
"\n"
"\n"
"\n"
"example:\n"
"        - self.list.getSelectedItem().setArt({ 'poster': 'poster.png', 'banner' : 'banner.png' }) \n");

Good to know then I can check it with the new doxygen documentation style.

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://github.com/xbmc/xbmc/pull/9368/files/1bf4fd27e56f068e21971e6b708ac66783bdcaba..cfe6c9ec0bf7660117e01a649fc6f004a430c016#r58816987>